front/app/components/CustomFieldsForm/generateYupSchema.ts

```diff
@@ -218,6 +218,21 @@ const generateYupValidationSchema = ({
           : number();
         break;
       }
+
+      case 'ranking': {
+        const numberOfOptions = question.options?.length;
+
+        // type guard, numberOfOptions should always be defined
+        schema[key] =
+          required && numberOfOptions !== undefined
+            ? array()
+                .of(string())
+                .min(numberOfOptions, formatMessage(messages.fieldRequired))
```

Comment:
Should we have separate messages for these scenarios?

Reply:
In this case, I don't think it's worth it, because in the interface is it actually impossible to achieve a state where you have a ranking that more than 0 but less than `numberOfOptions` selected options. The only case I can think of is when someone filled out the form, and then later the admin added new options, and you then go back to edit your response. In this case, the validation will just say that the thing is required instead of being more specific.